### PR TITLE
Allow TimeSync to use OSL LDAP accounts.

### PR DESCRIPTION
**Pull request by MorganEPatch**
@osuosl-cookbooks/chefs

### CODE REVIEW

**Comment by ramereth** [comment]
@LyonesGamer serverspec tests please?

**Comment by MorganEPatch** [comment]
I genuinely have no idea how we'd test this. I don't want to hook up tests to a production LDAP instance (much less stick any actual LDAP passwords in source control), and all this does is make it connect to the LDAP server.

**Comment by ramereth** [comment]
@LyonesGamer in this case, make sure the file it generates contains the settings you expect them to be. We don't need to actually test doing an ldap interaction here.

File: test/integration/default/serverspec/test_server_running_spec.rb [hunk 1]
```diff
@@ -34,3 +34,10 @@
     it { should exist }
   end
 end
+
+describe file('/etc/pm2/conf.d/timesync.json') do
+  it { should contain('"TIMESYNC_AUTH_MODULES": "[\"password\",\"ldap\"]"') }
+  it { should contain('"TIMESYNC_LDAP_URL": "ldaps://ldap.osuosl.org/"') }
+  it { should contain('"TIMESYNC_LDAP_SEARCH_BASE": "ou=People,dc=osuosl,dc=org"') }
```
Comment (ramereth) [inline test/integration/default/serverspec/test_server_running_spec.rb]: This line exceeds the 80char limit.

File: test/integration/default/serverspec/test_server_running_spec.rb [hunk 2]
```diff
@@ -34,3 +34,10 @@
...
+end
+
```
Comment (ramereth) [inline test/integration/default/serverspec/test_server_running_spec.rb]: Please remove extra newline at the end of the file.

File: test/integration/default/serverspec/test_server_running_spec.rb [hunk 3]
```diff
@@ -34,3 +34,11 @@
     it { should exist }
   end
 end
+
+describe file('/etc/pm2/conf.d/timesync.json') do
+  it { should contain('"TIMESYNC_AUTH_MODULES": "[\"password\",\"ldap\"]"') }
+  it { should contain('"TIMESYNC_LDAP_URL": "ldaps://ldap.osuosl.org/"') }
+  it do
+    should contain('"TIMESYNC_LDAP_SEARCH_BASE": "ou=People,dc=osuosl,dc=org"')
+  end
```
Comment (ramereth) [inline test/integration/default/serverspec/test_server_running_spec.rb:43]: None of these tests pass :(

**Comment by ramereth** [comment]
:+1: Good work!

**Comment by fahlmant** [comment]
This still fails for me on your newest commit

```
-----> serverspec installed (version 2.36.0)
       /opt/chef/embedded/bin/ruby -I/tmp/verifier/suites/serverspec -I/tmp/verifier/gems/gems/rspec-support-3.5.0/lib:/tmp/verifier/gems/gems/rspec-core-3.5.1/lib /opt/chef/embedded/bin/rspec --pattern /tmp/verifier/suites/serverspec/\*\*/\*_spec.rb --color --format documentation --default-path /tmp/verifier/suites/serverspec

       Port "8000"
         should be listening (FAILED - 1)

       Command "curl http://localhost/v0/"
         stdout
           should contain "Cannot GET /v0/" (FAILED - 2)
         exit_status
           should eq 0

       Port "80"
         should be listening

       install and start the Postgres database
         Service "postgresql"
           should be running
         Port "5432"
           should be listening

       create user and group
         User "timesync"
           should exist
         Group "timesync"
           should exist

       File "/etc/pm2/conf.d/timesync.json"
         should contain "\"TIMESYNC_AUTH_MODULES\":\"[\\\"password\\\", \\\"ldap\\\"]\""
         should contain "\"TIMESYNC_LDAP_URL\":\"ldaps://ldap.osuosl.org/\""
         should contain "\"TIMESYNC_LDAP_SEARCH_BASE\":\"ou=People,dc=osuosl,dc=org\""

       Command "curl -d '{"auth": {"type": "password", "username": "root", "password": "dLj+87o0fyYqPsbvlNuigpA/XFI9ThJAFjMMibEypDoJpAN0hMqBVs03fptaubFLCHBAh2mZ6lw0nlGwXhHMSQ"}}' -H "Content-Type: application/json" http://localhost/v0/login"
         stdout
           should match /^{\"token\":\s?\"[a-zA-Z0-9+\/=.]+\"}$/ (FAILED - 3)
         exit_status
           should eq 0
```

**Comment by ramereth** [comment]
@fahlmant have you run berks update?

**Comment by fahlmant** [comment]
:+1: that fixed it

**Comment by MorganEPatch** [comment]
!bump minor

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v0.3.0.  Have a nice day!

**Pull request merged**